Installation issue: mesa-glu with mesa as provider for gl

### Steps to reproduce the issue

When the provider for `gl` is `mesa` instead of `opengl`, the build of `mesa-glu` fails, because it cannot find `libGL`. According to `mesa-glu`'s `configure` script, one needs to add `--enable-osmesa` to build against a `mesa` backend.
```
source share/spack/setup-env.sh
spack env activate --temp
spack compiler find
spack external find --all
spack add mesa-glu%apple-clang ^mesa%apple-clang~llvm
spack spec
spack install -v
```

### Error message

`mesa-glu`'s `configure` call cannot find `libGL`:
```
...
1 error found in build log:
     112    checking for pkg-config... no
     113    checking for GL... no
     114    checking GL/gl.h usability... yes
     115    checking GL/gl.h presence... yes
     116    checking for GL/gl.h... yes
     117    checking for glBegin in -lGL... no
  >> 118    configure: error: GL required

See build log for details:
  /var/folders/gb/w8lys0xn3c35rbw_5vfq4mkxxgj2gv/T/heinzell/spack-stage/spack-stage-mesa-glu-9.0.1-pvyigbldnkromkf3ucdv43vb5b7tbsa2/spack-build-out.txt

==> Warning: Skipping external package: apple-libunwind@35.3%apple-clang@13.0.0 arch=darwin-monterey-x86_64/yiavq3f
==> Updating view at /var/folders/gb/w8lys0xn3c35rbw_5vfq4mkxxgj2gv/T/spack-lr2b95ck/.spack-env/view
```


### Information on your system

```
* **Spack:** 0.18.0.dev0 (d25933170c89576e1f53c3937af16ae8263dcad8)
* **Python:** 3.9.10
* **Platform:** darwin-monterey-x86_64
* **Concretizer:** clingo
```

### Additional information

[mesa_glu_logs.tar.gz](https://github.com/spack/spack/files/8492861/mesa_glu_logs.tar.gz)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/mesa-glu/package.py
+++ b/var/spack/repos/builtin/packages/mesa-glu/package.py
@@ -19,6 +19,12 @@ class MesaGlu(AutotoolsPackage):
 
     provides('glu@1.3')
 
+    def configure_args(self):
+        args = []
+        if self.spec['gl'].satisfies('mesa'):
+            args.append('--enable-osmesa')
+        return args
+
     @property
     def libs(self):
         for dir in ['lib64', 'lib']:

